frontend/alert/src/Timeline.js

```diff
@@ -0,0 +1,222 @@
+}
+
+function dateDivs(data, yoff){
+  var i;
+  let divs = [];
+  for (i=1; i<data.length; i++){
+    if(convertTime(data[i][0]["created_at"])[0]!==convertTime(data[i-1][0]["created_at"])[0]){
+      divs.push(<DateStyle offset={yoff[i]}>{convertTime(data[i][0]["created_at"])[0]}</DateStyle>);
+    }
+  }
+  console.log("date divs length is " + divs.length);
+  return divs;
+}
+
+function Timeline({courseCode, setTimeline}){
+  let [data, setData] = React.useState(null);
+  let [segLengths, setSegLengths] = React.useState(null);
+  let [yOffsets, setYOffsets] = React.useState(null);
+  let [loaded, setLoaded] = React.useState(false);
+  let [displayedCode, setDisplayedCode] = React.useState(null);
+  useEffect(()=>{
+      if(courseCode==null){
+        return;
+      }
+      setLoaded(false);
+      fetch(`https://penncourseplan.com/api/alert/statusupdate/${courseCode}`).then(res=>res.json()).then(result=>{
+        console.log(result)
+        result.sort((a,b)=>(a.created_at > b.created_at) ? 1 : -1);
+        let simplifiedData = result.reduce((ans, item, index) => { // preprocessing hte data
+          if(index==0){
+            return ans;
+          }
+          if(item["old_status"] == result[index-1]["old_status"]){
+            return ans;
+          }
+          if(item["old_status"] == "C" && item["new_status"] == "O"){
+            ans.push([item, "opened"]);
+          }
+          if(item["old_status"] == "O" && item["new_status"] == "C"){
+            ans.push([item, "closed"]);
+          }
+          return ans;
+        }, []).reverse()
+        setData(simplifiedData);
+        var i;
+        let segmentLengths = [];
+        for (i = 1; i < simplifiedData.length; i++){
+          segmentLengths.push(Math.round(20+5*Math.pow(1 + absoluteTime(result[i]["created_at"]) - absoluteTime(result[i-1]["created_at"]), 0.2)));
+        }
+        let yPositions = [];
+        yPositions[0] = 0;
+        for (i = 1; i < segmentLengths.length + 1; i++){
+          yPositions[i] = Math.round(yPositions[i-1] + segmentLengths[i-1]);
+        }
+        console.log(simplifiedData);
+        console.log(segmentLengths);
+        console.log(yPositions);
+        setSegLengths(segmentLengths);
+        setYOffsets(yPositions);
+        setLoaded(true);
+        setDisplayedCode(courseCode);
+      })
+    }
+  , [courseCode]);
+  return <TimelineContainer offScreen={courseCode==null || loaded==false}>
+            <MyButton onClick={()=>setTimeline(null)}><i className="fas fa-times"></i></MyButton>
+            <Heading>Alert History</Heading>
+            <LeftRight>
+                <Subheading>{displayedCode}</Subheading>
+                {loaded && data[0]["new_status"] === "O" ? <OpenBadge>Open</OpenBadge> : <ClosedBadge>Closed</ClosedBadge>}
+            </LeftRight>
+            <ScrollContainer>
+              { data && yOffsets ?
+                      <FlexRow>
+                        <div style={{width:"100px"}}>
```

Comment:
Probably should factor this out into a styled component, and also use rem over px.